frontend/src/components/App/Settings/DrawerModeButton.tsx

```diff
@@ -0,0 +1,49 @@
+import { FormControlLabel, Switch } from '@mui/material';
+import React from 'react';
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { setDetailDrawerEnabled } from '../../../redux/drawerModeSlice';
+import { useTypedSelector } from '../../../redux/reducers/reducers';
+
+export default function DrawerModeButton() {
+  const dispatch = useDispatch();
+  const { t } = useTranslation('translation');
+  // This will fix the problem of the project refreshing the state away from needed position
+
+  // DETAIL DRAWER MODE
+  const isDetailDrawerEnabled = useTypedSelector(state => state.drawerMode.isDetailDrawerEnabled);
+  const [isDrawerEnabled, changeDetailDrawerEnabled] = useState<any>(isDetailDrawerEnabled);
```

Comment:
isDrawerEnabled is duplicated state, use only the state from the store instead

Reply:
done!

Reply:
@vyncent-t

Reply:
done!